File: apps/renderer/src/platform/electron/events.ts
/**
 * @license
 * Copyright 2026 a7mddra
 * SPDX-License-Identifier: Apache-2.0
 */

import { platform } from "./index";
import type { PlatformEventMap } from "../shared.types";

export function listenTo<K extends keyof PlatformEventMap>(
  event: K,
  handler: (payload: PlatformEventMap[K]) => void,
) {
  return platform.listen(event as string, handler);
}

File: crates/squigit-brain/src/events.rs
// Copyright 2026 a7mddra
// SPDX-License-Identifier: Apache-2.0

use crate::provider::gemini::transport::types::GeminiEvent;

pub trait BrainEventSink: Send + Sync {
    fn emit(&self, channel_id: &str, event: GeminiEvent);
}

pub struct NoopEventSink;

impl BrainEventSink for NoopEventSink {
    fn emit(&self, _channel_id: &str, _event: GeminiEvent) {}
}
